modules/ROOT/pages/provisioning-raspberry-pi4.adoc

```diff
@@ -0,0 +1,188 @@
+
+
+## Installing FCOS and Booting via UEFI
+
+There is a UEFI firwmare implementation for the RPi4 (https://github.com/pftf/RPi4/[pftf/RPi4]) that attempts to make the RPi4 ServerReady (SBBC compliant) and allows you to pretend that the RPi4 is just like any other server hardware with UEFI.
+
+You can write the firwmare to a disk (USB or SD card) and then boot/install FCOS as you would on any bare metal server. However, the firmware files need to be on an SD card or USB disk and will take up either the SD card slot or a USB slot. Depending on your needs this may be acceptable or not. Depending on the answer you have a few options:
+
+- Separate UEFI Firmware Disk (aka "separate disk mode")
+- Combined Fedora CoreOS + UEFI Firmware Disk (aka "combined disk mode")
+
+These options are covered in the following sections. Regardless of which option you choose you'll want to consider if you need to either xref:#_uefi_firmware_changing_the_3g_limit[Change the 3G RAM limit] or xref:#_uefi_firmware_gpio_via_devicetree[Enable DeviceTree Boot].
+
+
+### UEFI: Separate UEFI Firmware Disk Mode
+
+In separate disk mode the UEFI firmware will take up either the SD card slot or a USB slot on your RPi4. Once the firmware disk is attached to the system you'll be able to follow the xref:bare-metal.adoc[bare metal install documentation] and pretend like the RPi4 is any other server hardware.
+
+To create a disk (SD or USB) with the firmware on it you can do something like:
+
+```nohighlight
+VERSION=v1.32  # use latest one from https://github.com/pftf/RPi4/releases
+UEFIDISK=/dev/sdX
+sudo mkfs.vfat $UEFIDISK
+mkdir /tmp/UEFIdisk
+sudo mount $UEFIDISK /tmp/UEFIdisk
+pushd /tmp/UEFIdisk
+sudo curl -LO https://github.com/pftf/RPi4/releases/download/${VERSION}/RPi4_UEFI_Firmware_${VERSION}.zip
+sudo unzip RPi4_UEFI_Firmware_${VERSION}.zip
+sudo rm RPi4_UEFI_Firmware_${VERSION}.zip
+popd
+sudo umount /tmp/UEFIdisk
+```
+
+Attaching this disk to your Pi4 you can now install FCOS as you would on any bare metal server.
+
+NOTE: The separate UEFI firmware disk will need to stay attached permanently for future boots to work.
+
+### UEFI: Combined Fedora CoreOS + UEFI Firmware Disk
+
+In combined disk mode the UEFI firmware will live inside the EFI partition of Fedora CoreOS, allowing for a single disk to be used for the UEFI firmware and FCOS.
+
+There are a few ways to achieve this goal:
+
+- Install Directly on RPi4
+- Prepare Pi4 Disk on Alternate Machine
+
+
+#### UEFI: Combined Disk Mode Direct Install
+
+When performing a direct install, meaning you boot (via the UEFI firmware) into the Fedora CoreOS live environment (ISO or PXE) and run `coreos-installer`, you can mount the EFI partition (2nd partition) of the installed FCOS disk after the install is complete and copy the UEFI firmware files over:
+
+```nohighlight
+UEFIDISK=/dev/mmcblkX or /dev/sdX
+FCOSDISK=/dev/sdY
+mkdir /tmp/mnt{1,2}
+sudo mount $UEFIDISK /tmp/mnt1
+sudo mount ${FCOSDISK}2 /tmp/mnt2
```

Comment:
Same here.

Reply:
+1 will do whatever we decide in https://github.com/coreos/fedora-coreos-docs/pull/336#discussion_r742372803